core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -220,14 +200,24 @@ private Resource toSlingResource(GridResource gridResource, ResourceResolver res
     private void generateCsvReport(Collection<GridResource> gridResources, ResourceResolver resourceResolver) {
         StopWatch stopWatch = StopWatch.createStarted();
         LOG.debug("Start CSV report generation");
+
         List<GridViewItem> gridViewItems = toSlingResourcesStream(gridResources, resourceResolver)
                 .map(resource -> resource.adaptTo(GridViewItem.class))
                 .filter(Objects::nonNull)
                 .collect(Collectors.toList());
-        byte[] csvContentBytes = CsvUtil.itemsToCsvByteArray(gridViewItems, this::printViewItemToCsv, CSV_COLUMNS);
-        LinkInspectorResourceUtil.removeResource(CSV_REPORT_PATH, resourceResolver);
-        LinkInspectorResourceUtil.saveFileToJCR(CSV_REPORT_PATH, csvContentBytes,
-                CsvUtil.CSV_MIME_TYPE, resourceResolver);
+
+        List<List<GridViewItem>> partitionGridViewItems = ListUtils.partition(gridViewItems, UI_ITEMS_LIMIT);
+        LinkInspectorResourceUtil.removeResource(CSV_REPORT_NODE_PATH, resourceResolver);
+        LinkInspectorResourceUtil.createNode(CSV_REPORT_NODE_PATH, resourceResolver);
+        LinkInspectorResourceUtil
+                .addParamToNode(CSV_REPORT_NODE_PATH, resourceResolver, CsvUtil.REPORTS_SIZE_PROPERTY_NAME, (long) partitionGridViewItems.size());
+
+        for (int i = 0; i < partitionGridViewItems.size(); i++) {
+            byte[] csvContentBytes = CsvUtil.itemsToCsvByteArray(partitionGridViewItems.get(i), this::printViewItemToCsv, null);
+            LinkInspectorResourceUtil.saveFileToJCR(LinkInspectorResourceUtil.buildResourcePathFromPageNumber(CSV_REPORT_NODE_PATH, i + 1), csvContentBytes,
```

Comment:
Consider committing resource changes not one-by-one but in a batch (remember how we used to do "migration scripts"?) As of now, if there are 1000 pages, there are 1000 `session.save()`-s